### PR TITLE
Issue 175 commit functionality from navbar

### PR DESCRIPTION
**Pull request by subahm**
- User should be able to commit from navbar

-Clicking on the commit button should open a commit modal

-User should be able to enter a message and commit their changes

### CODE REVIEW

File: app/misc/git.ts [hunk 1]
```diff
@@ -125,6 +125,9 @@ function stage() {
 
 function addAndCommit() {
   commitMessage = document.getElementById('commit-message-input').value;
+  if (commitMessage == null || commitMessage == "") {
+    commitMessage = document.getElementById('commit-message-input-navbar').value;
+  }
   if (commitMessage == null || commitMessage == "") {
     displayModal("Cannot commit without a commit message. Please add a commit message before committing");
     return;
```
Comment (jpstoldt) [inline app/misc/git.ts]: This sequence of if blocks would benefit from some comments explaining which commit message is used for the git commit and why.

File: app/components/header/header.component.html [hunk 2]
```diff
@@ -465,6 +465,26 @@ <h5 type="text">Upstream:</h5>
...
+  aria-hidden="true">
+  <div class="modal-dialog modal-sm">
+    <div class="modal-content" style=" width: 602px !important">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+        <h4 class="modal-title">Commit changes</h4>
+      </div>
+      <div class="modal-body" id="modal-text-box">
+        Please provide a commit message.
+        <input type="text" id="commit-message-input-navbar" class="form-control" style=" width: 554px !important" placeholder="Enter a commit message">
```
Comment (jpstoldt) [inline app/components/header/header.component.html]: I am wondering if we should pre-populate this textbox with the text entered for a commit message on the main screen?

**Comment by aeogunmuyiwa** [comment]
The changes work as described. Looks good to me

File: app/misc/git.ts (hunk 1 above)
Comment (subahm) [reply app/misc/git.ts]: Good idea. I have added some comments to explain the sequence of if statements and getting value of elements from 2 different ids

**Pull request merged**